website/content/docs/concepts/acl/auth-methods/oidc.mdx

```diff
@@ -0,0 +1,135 @@
+[OIDC](https://en.wikipedia.org/wiki/OpenID_Connect). This method allows
+authentication via a configured OIDC provider using the user's web browser.
+Initiate this method from the Nomad UI or the command line.
+
+## Prerequisites
+
+- General knowledge of [OIDC concepts](https://developer.okta.com/blog/2017/07/25/oidc-primer-part-1)
+- [Nomad Access Control List fundamentals][ACL Overview].
+
+Refer to [auth-method create] for the parameters required to create an OIDC auth-method.
+
+## JWT Verification
+
+Nomad uses OIDC discovery to verify JWT signatures against public
+keys from the issuer. Nomad first fetches keys from the OIDC
+Discovery URL during authentication and then applies OIDC
+validation criteria such as `iss` and `aud`.
+
+## OIDC Authentication
+
+Nomad includes two built-in OIDC login flows: the Nomad UI, and the CLI using
+[`nomad login`](/nomad/docs/commands/login).
+
+### Redirect URIs
+
+Properly setting redirect URIs is an important part of OIDC auth method
+configuration. You must configure this in both Nomad and the OIDC
+provider, and these configurations must align.
+
+Specify the redirect URIs for an auth method with the
+`AllowedRedirectURIs` parameter. The Nomad UI and CLI use different
+redirect URIs, so you need to configure one or both, depending on your
+installation.
+
+#### Nomad UI
+
+Logging in with the Nomad UI requires a redirect URI in one of the following forms:
+
+- `http://localhost:4649/ui/oidc/callback`
+- `https://{host:port}/ui/oidc/callback`
+
+The "host:port" must be correct for the Nomad agent serving the Nomad UI.
+
+#### CLI
+
+If you plan to support authentication that uses the
+`nomad login -method=<name>` command, you must configure a
+localhost redirect URI, which is usually
+`http://localhost:4649/oidc/callback`. Logins that use the CLI may
+specify a different host and listening port if needed. A URI with
+this host and port must match one of the configured redirected
+URIs. You must add these same "localhost" URIs to the provider
+as well.
+
+### OIDC Login
+
+#### Nomad UI
+
+1. Select one of the provider links in the Nomad homepage or navigate directly to `/ui/settings/tokens`.
+1. Click one of the buttons for your OIDC auth method of choice.
+1. Complete the authentication with the configured provider.
+
+#### CLI
+
+Execute the `nomad login -method=oidc` command to log in.
+
+```shell-session
+$ nomad login -method=oidc
+
+Complete the login via your OIDC provider. Launching browser to:
+
+    https://myco.auth0.com/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A4649%2Foidc%2Fcallback&client_id=r3qXc2bix9eF...
+```
+
+Your browser opens to the generated URL to complete the provider's login.
+Enter the URL manually if the browser does not automatically open.
+
+The callback listener defaults to `localhost:4649`. You may optionally
+pass a host and port to the callback listener with the
+`-oidc-callback-addr=<host:port>` flag. In this example, you specify custom host and port.
```

Comment:
I think a little more could be done to associate this callback addr to the CLI redirect URL